### PR TITLE
IFI-88 Creates the icon soy taglib

### PR DESCRIPTION
**Pull request by fernandosouza**
I would like to point that this version drops the `label` attribute of the JSP version for while. From what I understood, the JSP taglib uses the label to write inside the `<span>` element or to set the attribute `title` of the `<svg>` element. In the first case, it looks like another component the has an icon, for example, an `<button>` or `<a>`. In the second case, it tries to implement accessibility, but I'm not sure if this is the right approach. I'm doing a research about accessibility for icons.

### CODE REVIEW

**Comment by fernandosouza** [comment]
@brunobasto. Don't send forward this yet. I will update this pull.

**Comment by fernandosouza** [comment]
Hey, @carloslancha. We did changes in the TempalteRendererTag that reflect in the TranslationManagerTag. Can you take a look, please?

cc. @jbalsas

**Comment by fernandosouza** [comment]
ci:retest

File: modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/liferay-frontend.tld [hunk 1]
```diff
@@ -335,6 +335,30 @@
 		</attribute>
 	</tag>
 	<tag>
+		<description><![CDATA[Creates a UI component for managing icon.]]></description>
+		<name>icon</name>
+		<tag-class>com.liferay.frontend.taglib.servlet.taglib.soy.IconTag</tag-class>
+		<body-content>empty</body-content>
+		<attribute>
+			<description><![CDATA[CSS classes for styling the icon container and also define icons that was not defined as SVG.]]></description>
+			<name>elementClasses</name>
```
Comment (fernandosouza) [inline modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/liferay-frontend.tld]: @brunobasto, I would like to highlight this attribute name here `elementClasses`. I noticed that taglibs have used `cssClass` for this attribute type. My idea of using `elementClass` is to maintain consistency between MetalJS and soy taglibs. On the other hand, I know that soy taglibs should have as little impact as possible on the migration process. I'm not sure if this kind of change, even small, worth. What do you think?

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by fernandosouza** [comment]
Hey, guys. I've labelled commits with the wrong IFI reference. I'm going to rebase.

**Comment by jbalsas** [comment]
Hey @fernando, we can't commit anything under IFI issues. Those are only for our team, to track what we're doing.

To land code in portal, you need to create an LPS (story, task or bugs) and use it on the commits. You can link the LPS to the IFI for traceability purposes.

**Comment by brunobasto** [comment]
Hey @jbalsas, @fernandosouza is gonna resend. Other than that, we removed some attributes like Fernando mentioned above. It's basically similar to what we have in Clay and we could probably remove the .soy and .es files once Clay comes to portal. Is that ok?

**Comment by fernandosouza** [comment]
Hey, @jbalsas. Would you prefer I close this PR and send it again or rebase renaming those commits is ok?

**Comment by fernandosouza** [comment]
I think it's better closing this pull in order to update the branch reference.

**Comment by fernandosouza** [comment]
Resent in https://github.com/brunobasto/liferay-portal/pull/839

**Pull request closed without merging**